subjects/localhost/README.md

```diff
@@ -0,0 +1,162 @@
+## Localhost
+
+The Hypertext Transfer Protocol (HTTP) is an application layer protocol in the Internet protocol suite model for distributed, collaborative, hypermedia information systems.
+HTTP is the foundation of data communication for the World Wide Web. Hypertext documents include hyperlinks to other resources that the user can easily access, for example, by a mouse click or tap on the screen in a web browser.
+
+### Instructions
+
+#### Server
+
+The project can be written in one of these languages [Go,C++,C...]
+
+```console
+$ ./localhost [CONFIG FILE]
+Lisning on  [IP]:[PORT]
+Lisning on  [IP]:[PORT2]
+```
+- Your server must not crash for any reason.
+- Any request to your server should never hang forever.
+- Your server can listen on multiple servers and ports at the same time without any conflict 
+- You must use the single process and no treads.
+- You must use `exec` functions only in `CGI` part
+- Your server must be compatible with the last version of your chosen browser 
+- You can use `NGINX` to compare headers and answer behaviors with your server
+- You must manage at least [`GET`, `POST`, `DELETE`] methods.
+- You must handle customization errors **(Binding error, default page error...)**
+- You must create error page at least for [400,311,403,404,405,413]
+- You must execute CGI based on certain file extensions such us [`.php`,`.py`,...]
+- The `CGI` should run in the correct directory for relative path file access.
+- You must use the enverement paramatre of cgi. 
+- Your server must receive a request from the browser and send a response using the HTTP header and body
+- Your server must call `select` function (or equivalent function) one time only
+- You must manage chunked and unchunked request.
+- You must set the right status for each response
```

Comment:
- This list could profit by having a sub-title of its own.
- All points about CGI should be reorganized so they are all together and the most high level appear first, the most detailed and low level last.
- Some points end with `.`, some not. I'm fine with both approaches as soon as it is consistent, try to see on other subjects how lists are formatted. My personal opinion is to end it with a `.` since technically it is still a finite sentence.
- It could be nice to add a point to inform the student that for supported features we expect the behavior to be HTTP 1.1 compliant (I think this is the standard followed by nginx)

The following paragraph too could have a subtitle.